### PR TITLE
Cleanup rust

### PR DESCRIPTION
**Pull request by Sciencentistguy**
- Borrow command + trail instead of creating a new Vec
- Use the new `std::process:ExitCode` api instead of calling `std::process:exit()`
- Produce better error messages on invalid picker output

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -67,24 +69,29 @@ fn main() {
...
-    let attrs: Vec<&str> = std::str::from_utf8(&attrs)
+    let attrs: Vec<_> = std::str::from_utf8(&attrs)
         .expect("fail")
         .trim()
         .split('\n')
         .collect();
 
-    let choice = if attrs.len() != 1 {
-        pick(&args.picker, attrs)
+    dbg!(&attrs);
```
Comment (Artturin) [inline src/main.rs:81]: leftover from debugging?

Comment (Artturin) [reply src/main.rs:81]: if you'd like you can use some log crate and a `--verbose` option

**Comment by Artturin** [comment]
Thank you

File: src/main.rs (hunk 1 above)
Comment (Sciencentistguy) [reply src/main.rs:81]: Whoops, yeah i left that in by accident. I'll remove it.

**Comment by Artturin** [comment]
Been running this for 2 days and so far no issues

Always nice to see better rust

**Pull request merged**